src/Toolkit/PubSub/V1/PublisherApi.php

```diff
@@ -0,0 +1,673 @@
+        $clientConfig = json_decode($clientConfigJsonString, true);
+        $this->defaultCallSettings =
+                CallSettings::load(
+                    'google.pubsub.v1.Publisher',
+                    $clientConfig,
+                    $options['retryingOverride'],
+                    GrpcConstants::getStatusCodeNames(),
+                    $options['timeoutMillis']
+                );
+
+        $this->scopes = $options['scopes'];
+
+        $generatedCreateStub = function ($hostname, $opts) {
+            return new PublisherClient($hostname, $opts);
+        };
+        $createStubOptions = [];
+        if (!empty($options['sslCreds'])) {
+            $createStubOptions['sslCreds'] = $options['sslCreds'];
+        }
+        $this->grpcBootstrap = GrpcBootstrap::defaultInstance();
+        $this->stub = $this->grpcBootstrap->createStub(
+            $generatedCreateStub,
+            $options['serviceAddress'],
+            $options['port'],
+            $createStubOptions
+        );
+    }
+
+    /**
+     * Creates the given topic with the given name.
+     *
+     * Sample code:
+     * <pre><code>
+     * try {
+     *     $publisherApi = new PublisherApi();
+     *     $formattedName = PublisherApi::formatTopicName("[PROJECT]", "[TOPIC]");
+     *     $response = $publisherApi->createTopic($formattedName);
+     * } finally {
+     *     if (isset($publisherApi)) {
+     *         $publisherApi->close();
+     *     }
+     * }
+     * </code></pre>
+     *
+     * @param string $name         The name of the topic. It must have the format
+     *                             `"projects/{project}/topics/{topic}"`. `{topic}` must start with a letter,
+     *                             and contain only letters (`[A-Za-z]`), numbers (`[0-9]`), dashes (`-`),
+     *                             underscores (`_`), periods (`.`), tildes (`~`), plus (`+`) or percent
+     *                             signs (`%`). It must be between 3 and 255 characters in length, and it
+     *                             must not start with `"goog"`.
+     * @param array  $optionalArgs {
+     *                             Optional. There are no optional parameters for this method yet;
+     *                             this $optionalArgs parameter reserves a spot for future ones.
+     *                             }
+     * @param array  $callSettings {
+     *                             Optional.
+     *
+     *    @var Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for this call. If present, then
+     *          $timeout is ignored.
+     *    @var int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     *
+     * @return google\pubsub\v1\Topic
+     *
+     * @throws Google\GAX\ApiException if the remote call fails
+     */
+    public function createTopic($name, $optionalArgs = [], $callSettings = [])
```

Comment:
We considered it but decided it might be best to keep parameters specific to API methods separate from call settings, which are universal. Someone might want to hold a call settings object which they keep re-using, potentially in multiple API methods, where optional args are specific to an RPC.